Fix `flutter stop` to stop the right Android activity.

mpcomplete opened the pull request:
Previously we'd always stop org.domokit.sky.shell. If an
AndroidManifest.xml exists, we'll use the activity specified in there
instead.

Fixes https://github.com/flutter/flutter/issues/1651

devoncarew commented on packages/flutter_tools/lib/src/application_package.dart:
```diff
@@ -44,6 +48,36 @@ class AndroidApk extends ApplicationPackage {
+    Iterable<xml.XmlElement> manifests = document.findElements('manifest');
+    if (manifests.isEmpty)
+      return null;
+    String id = manifests.toList()[0].getAttribute('package');
```
Instead of `manifests.toList()[0]`, you could use `manifests.first` here.

mpcomplete replied:
Thanks! Done

eseidelGoogle commented:
Eventually we should probably remove these defaults all together?

devoncarew commented on packages/flutter_tools/lib/src/application_package.dart:
```diff
@@ -44,6 +48,36 @@ class AndroidApk extends ApplicationPackage {
     assert(launchActivity != null);
   }
+
+  // Creates a new ApplicationPackage from the Android manifest.
```
nit: you can use `///` (dartdoc) here as it's api documentation

devoncarew commented:
lgtm

mpcomplete committed:
Fix `flutter stop` to stop the right Android activity.

Previously we'd always stop org.domokit.sky.shell. If an
AndroidManifest.xml exists, we'll use the activity specified in there
instead.
